### ISSUE TITLE
Table Inserts monitoring using ProfileEvent_InsertedRows

### ISSUE THREAD

**Issue by bun4uk** [ClickHouse/ClickHouse#21781]
**Use case**
We would like to monitor insert events into our ClickHouse cluster to be sure that our Data Lake works correctly. 
Currently, we can check total rows inserts events using `system.metric_log` table and `ProfileEvent_InsertedRows` column. But we are looking at the way how to monitor each table separately. 

**Describe the solution you'd like**
It would be nice to have possibility to get `ProfileEvent_InsertedRows` metric and group it by table_name

**Describe alternatives you've considered**
Currently, we created our own solution based on a bunch of materialized views for each table that we insert. These materialized views write data to a special table with the next structure: 
```
┌─name─────┬─type───────────────────────────────────────────────────────────────────────────┬─default_type─┬─default_expression─┬─comment─┬─codec_expression─┬─ttl_expression─┐
│ date     │ Date                                                                           │              │                    │         │                  │                │
│ ts       │ DateTime                                                                       │              │                    │         │                  │                │
│ table    │ String                                                                         │              │                    │         │                  │                │
│ rows     │ Int64                                                                          │              │                    │         │                  │                │
│ hostname │ String                                                                         │              │                    │         │                  │                │
│ source   │ Enum8('kafka' = 1, 'mysql' = 2, 'matview' = 3, 'aggregation' = 4, 'other' = 5) │ DEFAULT      │ 'other'            │         │                  │                │
└──────────┴────────────────────────────────────────────────────────────────────────────────┴──────────────┴────────────────────┴─────────┴──────────────────┴────────────────┘
```

**Additional context**
Based on these metrics we created dashboards and are working on alerting system.
![CleanShot 2021-03-16 at 14 07 17@2x](https://user-images.githubusercontent.com/14631690/111307879-aea68e00-8662-11eb-8b3a-a62b76c5ef86.png)

**Comment by igor9292** [comment]
+

**Comment by bTokman** [comment]
++

**Comment by markobt** [comment]
+

**Comment by Levsha128** [comment]
+

**Comment by gvadrian** [comment]
Very nice issue

**Comment by Talapius** [comment]
+

**Comment by tony10010110** [comment]
+

**Comment by akuzm** [comment]
GitHub has a reactions feature for expressing your interest in the issue, please use it. Posting ten different comments with a dot is harmful, because it hides the discussion and generates meaningless notifications. I have temporarily locked the discussion on this issue.

**Comment by filimonov** [comment]
Did you consider selecting needed data from `system.tables`?

**Comment by bun4uk** [comment]
> Did you consider extracting data from `system.tables`?

Yes. I can have a total count of rows for each table from `system.tables`.  But to have such dashboard, that I attached, I have to do periodical snapshots of system.tables table and count delta between current and previous table state. It requires creating some cron jobs that will do the process. So the overall system would be more complex than it could be.

**Comment by Slach** [comment]
@bun4uk you can try to use https://github.com/lmangani/clickhouse-metrics-prometheus to allow export custom SQL queries as Prometheus metrics

**Comment by alexey-milovidov** [comment]
Also it's possible to extract the relevant data from `system.query_log`.
It has `tables` field, `ProfileEvents` and `query_kind`.

**Comment by bun4uk** [comment]
@alexey-milovidov from which CH version these fields exist?
We have 20.10 and I don't see them

**Comment by Slach** [comment]
@bun4uk according to git blame 
https://github.com/ClickHouse/ClickHouse/commit/272311f6bdccbc9832fd63b7f230136010d2547a
available from 20.13
